src/argilla/client/feedback/schemas/questions.py

```diff
@@ -153,9 +154,34 @@ def update_settings(cls, values: Dict[str, Any]) -> Dict[str, Any]:
             ]
         if isinstance(values.get("labels"), list):
             values["settings"]["options"] = [{"value": label, "text": label} for label in values.get("labels")]
-        values["settings"]["visible_options"] = values.get(
-            "visible_labels"
-        )  # `None` is a possible value, which means all labels are visible
+        visible_options = values.get("visible_labels")
+        total_options = len(values["settings"]["options"])
+        if visible_options and visible_options > total_options:
+            if total_options >= 3:
+                warnings.warn(
+                    f"`visible_labels={visible_options}` is greater than the total number"
+                    f" of labels ({total_options}), so it will be set to `{total_options}`.",
+                    UserWarning,
+                    stacklevel=1,
+                )
+                visible_options = total_options
+            else:
+                warnings.warn(
+                    f"`labels={values.get('labels')}` has less than 3 labels, so `visible_labels`"
+                    " will be set to `None`, which means that all the labels will be visible.",
+                    UserWarning,
+                    stacklevel=1,
+                )
+                visible_options = None
+        if not visible_options and total_options > 20:
+            warnings.warn(
+                "Since `visible_labels` has not been provided or is `None`, and the total number"
+                " of labels is greater than 20, `visible_labels` will be set to `20`.",
```

Comment:
As I said before, as a user is weird to me to change the `visible_labels` value when I explicitly set it to `None` which means that I want to show all the labels indiferently of the number of labels I have.